src/tests/src/solver/optim/api/testApiOrtoolsLinearProblem.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+#define BOOST_TEST_MODULE test optim api ortools
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <ortools/linear_solver/linear_solver.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <antares/solver/optim/ortoolsImpl/linearProblem.h>
+
+using namespace Antares::Solver::Optim;
+
+struct Fixture
+{
+    ~Fixture() = default;
+
+    Fixture()
+    {
+        pb = std::make_unique<OrtoolsImpl::OrtoolsLinearProblem>(false, "sirius");
+    }
+
+    std::unique_ptr<OrtoolsImpl::OrtoolsLinearProblem> pb;
+
+    void createProblemMaximize();
+};
+
+void Fixture::createProblemMaximize()
+{
+    auto* a = pb->addNumVariable(0, 1, "a");
+    auto* b = pb->addNumVariable(0, 1, "b");
+    auto* c = pb->addConstraint(1, 1, "c");
+
+    c->setCoefficient(a, 1);
+    c->setCoefficient(b, 1);
+
+    pb->setObjectiveCoefficient(a, 1);
+    pb->setObjectiveCoefficient(b, 1);
+    pb->setMaximization();
```

Comment:
`pb->setMaximization();` : 
Do we need to "maximize" for a problem that small ?

Reply:
I'm not sure I understand, can you elaborate ?

Reply:
> I'm not sure I understand, can you elaborate ?

I mean : whatever means "maximizing" a problem, do we need to do it in a unit test, that involves a tiny linear problem ?
If we remove the maximization, do we really loose something ?

Reply:
Maximize is meant to tell the solver something along the lines of

> Find x such that b.x is **maximal** under the given set of constraints & bounds

By calling `setMaximization`, we just set the sign (+/-) of the objective function `x -> b.x`.

It is not intended to expand the problem, whatever that means.